lib/foreman_tasks/event_subscribers_extensions.rb

```diff
@@ -0,0 +1,11 @@
+module ForemanTasks
+  module EventSubscribersExtensions
+    def all_observable_events
+      @all_observable_events ||=
+        begin
+          actions = ::Dynflow::Action.descendants.select { |klass| klass <= ::Actions::ObservableAction }.map(&:namespaced_event_names)
+          (super + actions).flatten.uniq
+        end
+    end
+  end
+end
```

Comment:
Can you move this to `engine.rb` and use the newly created `extend_observable_events`? You will need to checkout https://github.com/theforeman/foreman/pull/8284 for that.